oauth2/oauth2_auth_code_test.go

```diff
@@ -977,6 +1150,103 @@ func TestAuthCodeWithDefaultStrategy(t *testing.T) {
 	})
 }
 
+func assertCreateVerifiableCredential(t *testing.T, reg driver.Registry, nonce string, accessToken *oauth2.Token, alg jose.SignatureAlgorithm) {
+	// Build a proof from the nonce.
+	pubKey, privKey, err := josex.NewSigningKey(alg, 0)
+	require.NoError(t, err)
+	pubKeyJWK := &jose.JSONWebKey{Key: pubKey, Algorithm: string(alg)}
+	proofJWT := createVCProofJWT(t, pubKeyJWK, privKey, nonce)
+
+	// Assert that we can fetch a verifiable credential with the nonce.
+	verifiableCredential, _, err := createVerifiableCredential(t, reg, accessToken, &hydraoauth2.CreateVerifiableCredentialRequestBody{
+		Format: "jwt_vc_json",
+		Types:  []string{"VerifiableCredential", "UserInfoCredential"},
+		Proof: &hydraoauth2.VerifiableCredentialProof{
+			ProofType: "jwt",
+			JWT:       proofJWT,
+		},
+	})
+	require.NoError(t, err)
+	assertVerifiableCredentialContainsPublicKey(t, reg, verifiableCredential, pubKeyJWK)
+}
+
+func assertVerifiableCredentialContainsPublicKey(t *testing.T, reg driver.Registry, vc *hydraoauth2.VerifiableCredentialResponse, pubKeyJWK *jose.JSONWebKey) {
+	ctx := context.Background()
+	token, err := jwt.Parse(vc.Credential, func(token *jwt.Token) (interface{}, error) {
+		return x.Must(reg.OpenIDJWTStrategy().GetPublicKey(ctx)).Key, nil
+	})
+	require.NoError(t, err)
+	pubKeyRaw, err := pubKeyJWK.MarshalJSON()
+	require.NoError(t, err)
+	expectedID := fmt.Sprintf("did:jwk:%s", base64.RawURLEncoding.EncodeToString(pubKeyRaw))
+	require.Equal(t, expectedID, token.Claims.(jwt.MapClaims)["vc"].(map[string]any)["credentialSubject"].(map[string]any)["id"])
+}
+
+func createVerifiableCredential(
+	t *testing.T,
+	reg driver.Registry,
+	token *oauth2.Token,
+	createVerifiableCredentialReq *hydraoauth2.CreateVerifiableCredentialRequestBody,
+) (vcRes *hydraoauth2.VerifiableCredentialResponse, vcErr *fosite.RFC6749Error, err error) {
```

Comment:
This signature is kinda wonky. We never assign to `vcErr` in the body of the function.

The idea is to return either a successful VerifiableCredentialResponse, or an RFC6749Error, correct?

In that case, we should remove the third`err error` return value and rely on `require.NoError` inside the function.

Reply:
Thanks, that'd work!